quantum/mousekey.c

```diff
@@ -17,19 +17,29 @@
 
 #include <stdint.h>
 #include <string.h>
+#include <math.h>
 #include "keycode.h"
 #include "host.h"
 #include "timer.h"
 #include "print.h"
 #include "debug.h"
 #include "mousekey.h"
+#include "lib/lib8tion/lib8tion.h"
 
 static inline int8_t times_inv_sqrt2(int8_t x) {
-    // 181/256 is pretty close to 1/sqrt(2)
-    // 0.70703125                 0.707106781
+#ifdef MOUSEKEY_PRECISE_DIAGONAL_MOVE
+    // 46341/65536 is very near to 1/sqrt(2)
+    // 0.70710754                  0.707106781
+    return (x < 0 ? -1 : 1) * scale16(x < 0 ? -x : x, 46341);
+#elif defined MOUSEKEY_PRECISE_DIAGONAL_MOVE_2
+    return round((x < 0 ? -1 : 1) * scale16(x < 0 ? -x : x, 46341));
+#else
+    // 181/256 is also pretty close to 1/sqrt(2)
+    // 0.70703125
```

Comment:
```suggestion
#ifdef MOUSEKEY_PRECISE_DIAGONAL_MOVE
    // 46341/65536 is very close to 1/sqrt(2)
    // 0.70710754                  0.707106781
    return round((x < 0 ? -1 : 1) * scale16(x, 46341));
#else
    // 181/256 is pretty close to 1/sqrt(2)
    // 0.70703125                 0.707106781
```

The first parameter for `scale16` is an unsigned integer so `-x` shouldn't be require:
```c
LIB8STATIC uint16_t scale16(uint16_t i, fract16 scale)
```